src/microbe_stage/BackgroundPlane.cs

```diff
@@ -0,0 +1,263 @@
+using System;
+using Godot;
+
+/// <summary>
+///   Manages the microbe background plane, optionally applies blur.
+/// </summary>
+public partial class BackgroundPlane : Node3D, IGodotEarlyNodeResolve
+{
+    private readonly StringName blurAmountParameter = new("blurAmount");
+    private readonly StringName textureAlbedoParameter = new("textureAlbedo");
+    private readonly StringName worldPositionParameter = new("worldPos");
+    private readonly StringName lightLevelParameter = new("lightLevel");
+    private readonly StringName distortionStrengthParameter = new("distortionFactor");
+
+    [Export]
+    private NodePath? blurPlanePath;
+
+    [Export]
+    private NodePath blurColorRectPath = null!;
+
+    [Export]
+    private NodePath backgroundPlanePath = null!;
+
+#pragma warning disable CA2213
+
+    /// <summary>
+    ///   Background plane that is moved farther away from the camera when zooming out
+    /// </summary>
+    private Node3D backgroundPlane = null!;
+
+    private Node3D blurPlane = null!;
+
+    private GpuParticles3D? backgroundParticles;
+
+    private ShaderMaterial spatialBlurMaterial = null!;
+
+    private ShaderMaterial canvasBlurMaterial = null!;
+
+    private ShaderMaterial currentBackgroundMaterial = null!;
+
+    private SubViewport subViewport1 = null!;
+
+    private SubViewport subViewport2 = null!;
+#pragma warning restore CA2213
+
+    public bool NodeReferencesResolved { get; private set; }
+
+    public float PlaneOffset
```

Comment:
Why is this not the direct offset but 15 is added on top? Isn't this quite unexpected, like for a variable like "plane offset" I'd expect it to set the plane offset, not plane offset with extra modifier on top...